corehq/apps/data_dictionary/models.py

```diff
@@ -0,0 +1,38 @@
+from django.db import models
+
+PROPERTY_TYPE_CHOICES = (
+    ('Date', 'Date'),
+    ('Plain', 'Plain'),
+    ('Number', 'Number'),
+    ('Select', 'Select'),
+    ('Integer', 'Integer'),
+    ('', 'No Type Currently Selected')
+)
+
+
+class CaseType(models.Model):
+    domain = models.CharField(max_length=255)
```

Comment:
Should add `default=None` to force integrity errors if no domain is supplied otherwise it will save with domain=''. Same for other non-nullable char fields